client/state/selectors/has-unseen-notifications.js

```diff
@@ -0,0 +1,25 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+
+import getUnseenCount from 'state/selectors/get-notification-unseen-count';
+import { getCurrentUser } from 'state/current-user/selectors';
+
+export const hasUnseenNotifications = getUser => state => {
```

Comment:
I suppose why are we passing `getUser`? Is this to make testing easier?

Unless we have a specific use case, maybe let's keep a simpler interface, and pass through a larger data object in the test
```jsx
export default function hasUnseenNotifications( state ) {

   const unseenCount = getUnseenCount( state );
   if ( null !== unseenCount ) {
	   return unseenCount > 0;
   }
   const user = getCurrentUser( state ) {
   if ( null !== user ) {
       return !! user.has_unseen_notes;
   }
   return null;
}
//... in the test file
// as a reader it's easier for me to grok what we mean, by passing needed global state:
test( 'falls back to user info', () => {
	expect( hasUnseenNotifications( 
        {
           notificationsUnseenCount: null,
           currentUser: {
			id: 73705554,
            has_unseen_notes: false
		    }
       } 
   ) ).toBe( false );
	expect( hasUnseenWithUserUnseen(
        {
            notificationsUnseenCount: null,
           currentUser: {
			id: 73705554,
            has_unseen_notes: false
		    }
       } 
   ) ) ).toBe( true );
} );

```

Reply:
Overall code changes here look good 👍 Feel free to 🚢 after you clarify this case ^^

Reply:
haha, sorry @gwwar - I recommended that we use the `getUser` mock since I didn't want to test the state system of `getUser` _and_ of the unseen count in this unit.

the sad thing here is that all we really care about is a property on the current user object but what we end up having to do is understand and deeply couple into the specifics of how that is stored - what I was trying to avoid by injecting it.

how strongly do you feel about this? if you are okay with it, do you think a comment on _why_ it's there would resole the concern?

we used this to discuss a few ideas about testing so it's not like we're inflexible here either.

Reply:
> the sad thing here is that all we really care about is a property on the current user object 

Selectors can access anything in the global state tree, so this coupling in tests, I don't particularly mind. If we end up modifying the user tree structure, this test is easy to update, and also marks that we care about a specific user prop. (Sometimes we don't know what we can trim from an endpoint response, and for example when we trimmed down the site object, we just had to guess and see what broke later.)

I won't block here, but at least leave a comment that it's meant for testing purposes only. When I see an interface like that I assume we have use cases for them, where we swap actual `getUser` implementations.